packages/react-query/src/useBaseQuery.ts

```diff
@@ -29,29 +29,43 @@ export function useBaseQuery<
   >,
   Observer: typeof QueryObserver,
 ) {
+  // Duc - we need to create a queryClient and reuse it per query
+
+  // { context: options.context }
+  // Duc - why do we need context here?
+  // Duc - how to implement this behavior in angular
+  // Duc - FROM the document: Required, but only if no default query function has been defined See Default Query Function for more information.
+  // https://tanstack.com/query/v4/docs/reference/useQuery?from=reactQueryV3&original=https://react-query-v3.tanstack.com/reference/useQuery
+  // maybe we can skip this for angular e.g. the default query function is required
   const queryClient = useQueryClient({ context: options.context })
+  // Duc - skip this in MVP
   const isRestoring = useIsRestoring()
```

Comment:
the check for restoring is needed to make sure that we are not triggering queries while an async persister is potentially still restoring data from an external location into the query cache.

You'll need something like this if the persist-query-client plugins are supported.